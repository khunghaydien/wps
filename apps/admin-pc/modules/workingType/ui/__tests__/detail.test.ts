// @ts-nocheck
import snapshotDiff from 'snapshot-diff';

import reducer, {
  actions,
  INITIALIZE,
  initialState,
  SET_SELECTED_HISTORY_ID,
  State,
} from '../detail';

describe('reducer()', () => {
  test('@@INIT', () => {
    // @ts-ignore
    const next = reducer(undefined, { type: '@@INIT' });
    expect(snapshotDiff({}, next)).toMatchSnapshot();
  });
  test(INITIALIZE, () => {
    // Arrange
    const prev: State = {
      // @ts-ignore
      selectedIndex: 0,
    };
    // Execute
    const next = reducer(prev, actions.initialize());
    // Assert
    expect(snapshotDiff(prev, next)).toMatchSnapshot();
  });
  test(SET_SELECTED_HISTORY_ID, () => {
    // Arrange
    const prev = initialState;
    // Execute
    // @ts-ignore
    const next = reducer(prev, actions.setSelectedHistoryId(0));
    // Assert
    expect(snapshotDiff(prev, next)).toMatchSnapshot();
  });
});
