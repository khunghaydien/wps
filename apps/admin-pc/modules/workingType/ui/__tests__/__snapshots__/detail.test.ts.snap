// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer() @@INIT 1`] = `
"Snapshot Diff:
- First value
+ Second value

- Object {}
+ Object {
+   \\"selectedHistoryId\\": \\"\\",
+ }"
`;

exports[`reducer() ADMIN-PC/MODULES/WORKING_TYPE/UI/DETAIL/INITIALIZE 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"selectedIndex\\": 0,
+   \\"selectedHistoryId\\": \\"\\",
  }"
`;

exports[`reducer() ADMIN-PC/MODULES/WORKING_TYPE/UI/DETAIL/SET_SELECTED_HISTORY_ID 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"selectedHistoryId\\": \\"\\",
+   \\"selectedHistoryId\\": 0,
  }"
`;
